app/components/qr-scanner/qr-scanner.tsx

```diff
@@ -0,0 +1,83 @@
+import '~/components/qr-scanner/qr-scanner.css';
+import Scanner from 'qr-scanner';
+import { useEffect, useRef, useState } from 'react';
+import { useAnimatedQRDecoder } from '~/lib/cashu/use-animated-qr-decoder';
+
+const AnimatedScanProgress = ({ progress }: { progress: number }) => {
```

Comment:
On the note of conventions. Sometimes I like to separate out things like this just because it helps with readability even though it will probably never be used outside of this file. 1. Do you agree with that? 2. Should these types of components that aren't the primary export go to the bottom of the file?

Reply:
1 I do like that. I often do the same.

2 I am fine both ways. I don't now why but I usually put those before the main component (unlike in classes where I put public methods first which is a habit from my c#/dotnet days)

Reply:
Okay I will leave as is, I too usually put those before the main component. I guess with this case we are more interested in what will end up in the main component so its nice to see things like `AnimatedScanProgress` before looking at the main component